pallets/axelar-router/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+use cfg_traits::{
+	ethereum::EthereumTransactor,
+	liquidity_pools::{MessageReceiver, MessageSender},
+	PreConditions,
+};
+use cfg_types::{domain_address::DomainAddress, EVMChainId};
+use ethabi::{Contract, Function, Param, ParamType, Token};
+use fp_evm::{ExitError, PrecompileFailure, PrecompileHandle};
+use frame_support::{
+	pallet_prelude::*,
+	weights::{constants::RocksDbWeight, Weight},
+	BoundedVec,
+};
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+use precompile_utils::prelude::*;
+use sp_core::{H160, H256, U256};
+use sp_std::collections::btree_map::BTreeMap;
+
+/// Maximum size allowed for a byte representation of an Axelar EVM chain
+/// string, as found below:
+/// <https://docs.axelar.dev/dev/reference/mainnet-chain-names>
+/// <https://docs.axelar.dev/dev/reference/testnet-chain-names>
+const MAX_AXELAR_EVM_CHAIN_SIZE: u32 = 16;
+
+const MAX_SOURCE_CHAIN_BYTES: u32 = 128;
+// Ensure we allow enough to support a hex encoded address with the `0x` prefix.
+const MAX_SOURCE_ADDRESS_BYTES: u32 = 42;
+const MAX_TOKEN_SYMBOL_BYTES: u32 = 32;
+const MAX_PAYLOAD_BYTES: u32 = 1024;
+const EVM_ADDRESS_LEN: usize = 20;
+
+pub type ChainName = BoundedVec<u8, ConstU32<MAX_AXELAR_EVM_CHAIN_SIZE>>;
+
+/// Type to represent the kind of message received by Axelar
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum AxelarId {
+	Evm(EVMChainId),
+}
+
+/// Configuration for outbound messages though axelar
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct AxelarConfig {
+	/// Address of liquidity pool contract in the target chain
+	pub liquidity_pools_contract_address: H160,
+
+	/// Configuration for executing the EVM call.
+	pub domain: DomainConfig,
+}
+
+/// Specific domain configuration
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum DomainConfig {
+	Evm(EvmConfig),
+}
+
+/// Data for validating and executing the internal EVM call.
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct EvmConfig {
+	/// Associated chain id
+	pub chain_id: EVMChainId,
+
+	/// The address of the contract deployed in our EVM.
+	pub target_contract_address: H160,
+
+	/// The `BlakeTwo256` hash of the target contract code.
+	///
+	/// This is used during router initialization to ensure that the correct
+	/// contract code is used.
+	pub target_contract_hash: H256,
+
+	/// The values used when executing the EVM call.
+	pub fee_values: FeeValues,
+}
```

Comment:
Some fields have been merged with the previous `axelar_precompile` configuration. For example, `set_gateway()` previously set in `axelar_precompile` was the same as `liquidity_pools_contract_address` for the outbound router.

Anyways, maybe we need to polish it a bit when updating the ITs.